huggingface_pipelines/dataset.py

```diff
@@ -0,0 +1,194 @@
+
+        Note:
+            Ensure that the 'datasets' library is installed before calling this method.
+        """
+        from datasets import load_dataset
+
+        dataset_kwargs = self.get_dataset_kwargs()
+        dataset = load_dataset(**dataset_kwargs)
+
+        self.validate_world_size_and_rank()
+
+        if not self.streaming and self.world_size > 1:
+            dataset = dataset.shard(
+                num_shards=self.world_size, index=self.rank)
+
+        return self.post_process_dataset(dataset)
+
+    def get_dataset_kwargs(self) -> dict[str, Any]:
+        """
+        Returns the kwargs for load_dataset function.
+
+        This method prepares the keyword arguments used in the load_dataset function.
+
+        Returns:
+            dict[str, Any]: A dictionary of keyword arguments for load_dataset.
+        """
+        dataset_kwargs = {
+            "path": self.dataset_name,
+            "name": self.config,
+            "split": self.dataset_split,
+            "trust_remote_code": self.trust_remote_code,
+            "streaming": self.streaming
+        }
+        return {k: v for k, v in dataset_kwargs.items() if v is not None}
+
+    def validate_world_size_and_rank(self):
+        """
+        Validates world_size and rank.
+
+        Raises:
+            AssertionError: If world_size or rank are invalid.
+        """
+        assert self.world_size >= 1, f"Invalid world_size: {self.world_size}. It should be >= 1."
+        assert 0 <= self.rank < self.world_size, f"Invalid rank: {self.rank}. It should be between 0 and {self.world_size - 1}."
+
+    def post_process_dataset(self, dataset):
+        """
+        Performs any post-processing on the dataset.
+
+        This method can be overridden in subclasses to implement
+        dataset-specific post-processing.
+
+        Args:
+            dataset (datasets.Dataset): The loaded dataset.
+
+        Returns:
+            datasets.Dataset: The post-processed dataset.
+        """
+        return dataset
+
+    def with_overwrites(self, overwrites: DatasetOverwrites):
+        """
+        Creates a new instance with specified overwrites.
+
+        This method allows for the creation of a new configuration object
+        with some attributes overwritten, without modifying the original instance.
+
+        Args:
+            overwrites (DatasetOverwrites): A dictionary of attributes to overwrite.
+
+        Returns:
+            BaseDatasetConfig: A new instance with the specified overwrites applied.
+
+        Example:
+            new_config = config.with_overwrites({"dataset_split": "test", "world_size": 4})
+        """
+        return replace(self, **overwrites)
+
+
+@dataclass
+class TextDatasetConfig(DatasetConfig):
+    """
+    Configuration for text datasets.
+
+    This class inherits from BaseDatasetConfig and can be used for
+    text-specific dataset configurations.
+    """
+
+
+@dataclass
+class AudioDatasetConfig(DatasetConfig):
```

Comment:
they should live in their corresponding modules, but let's move them once it's merged !